clients/drcachesim/common/trace_entry.h

```diff
@@ -736,12 +736,15 @@ typedef enum {
         OFFLINE_FILE_TYPE_ARCH_ARM32 | OFFLINE_FILE_TYPE_ARCH_X86_32 |
         OFFLINE_FILE_TYPE_ARCH_X86_64, /**< All possible architecture types. */
     /**
+     * Instruction addresses filtered online.
      * Note: this file type may transition to non-filtered. This transition is indicated
-     * by the TRACE_MARKER_TYPE_FILTER_ENDPOINT marker. Each window (which is indicated by
-     * the TRACE_MARKER_TYPE_WINDOW_ID marker) starts out filtered.
-     * TODO: add a new file type for mixed traces.
+     * by the #TRACE_MARKER_TYPE_FILTER_ENDPOINT marker. Each window (which is indicated
+     * by the #TRACE_MARKER_TYPE_WINDOW_ID marker) starts out filtered. This applies to
+     * #OFFLINE_FILE_TYPE_DFILTERED also. Note that threads that created after the
+     * transition will also have this marker - right at the beginning.
      */
-    OFFLINE_FILE_TYPE_IFILTERED = 0x80,  /**< Instruction addresses filtered online. */
+    /* TODO i#6164: add a new file type for mixed traces. */
```

Comment:
Let's try to get this new filetype in not too long after this PR, to reduce the window where people might create these anomalous traces with mixed contents but no top-level indicator that will confuse analysis tools and to avoid anyone investing in building analyzers to handle such traces by dynamically discovering it when they could have waited for the new filetype.

Reply:
I thought that since the FILTER_ENDPOINT marker was added as part of #5675 a while ago, we have had these mixed traces for a while, so adding this new filtype was more of a nice to have feature.

Reply:
That is true; I guess it feels like it will be much easier and more common to create such traces though with a runtime option versus the offline transformation which seems like a rarely used avenue.  In terms of adding the new flag it has more value the sooner it is added.

Reply:
Please add a #6164 reference to the PR description where it says:

> new OFFLINE_FILE_TYPE_BIMODAL type to indicate that the file has both filtered and unfiltered entries

Reply:
> That is true; I guess it feels like it will be much easier and more common to create such traces though with a runtime option versus the offline transformation which seems like a rarely used avenue. In terms of adding the new flag it has more value the sooner it is added.

Okay, I'll try to get to this soon.

> Please add a #6164 reference to the PR description where it says:
> 
> > new OFFLINE_FILE_TYPE_BIMODAL type to indicate that the file has both filtered and unfiltered entries

Done.